include/model.php

```diff
@@ -842,7 +842,18 @@ protected function get_languages_from_taxonomies() {
 		}
 
 		// Restore the right order after the first `array_reverse()`.
-		$terms_by_slug = array_reverse( $terms_by_slug );
+		// Also keep the terms order with the 'language' taxonomy first.
+		$terms_by_slug = array_reverse(
+			array_map(
+				function( $term_by_slug ) {
+					if ( 0 === $term_by_slug['language']->term_group ) {
+						return $term_by_slug;
+					}
+					return array_reverse( $term_by_slug );
+				},
+				$terms_by_slug
+			)
+		);
```

Comment:
These `array_reverse()` things are getting too complex. I propose to remove all of them and correctly sort terms directly in the database query by replacing [the orderby](https://github.com/polylang/polylang/blob/ca8c61187157d227b43c916c8b15bcb9c49d0340/include/model.php#L747) with something like:
```php
return sprintf( 'tt.taxonomy = "language", %1$s.term_group, %1$s.term_id', $matches['alias'] );
```

Reply:
Ha, I see what you're trying to do. That may simplify things indeed.
However, I'm not sure if you want `language` terms first or last here, but they will be last with this.
To get them first, I did that:
```sql
ORDER BY tt.taxonomy = 'language' DESC, t.term_group ASC, t.term_id ASC
```

Reply:
🤔 it can' work as we expected. 

Some tests as `Model_Test::test_languages_list_order()` confirm we break the languages list order instead of the solution I proposed at the beginning.

I will explain with more detailed explanations.

⚠we can't put `ASC` or `DESC` at the end of the orderby clause we modify because `get_terms()` add it with `ASC` value by default. If we want to change it, we need to use the `order` parameter passed to `get_terms()`
It is also useless to use ASC because it's the MySQL default.
See 
- https://github.com/WordPress/WordPress/blob/6.2/wp-includes/class-wp-term-query.php#L454 
- https://github.com/WordPress/WordPress/blob/6.2/wp-includes/class-wp-term-query.php#L738
Default values are set by the `parse_query()` call See https://github.com/WordPress/WordPress/blob/6.2/wp-includes/class-wp-term-query.php#L351

Reply:
So I propose a mixed solution.
A part of the code I proposed but simplified because the `get_terms()` is also order by `tt.taxonomy = "language"` which puts all the `language` terms at the beginning of the result. Thus we always have to reverse the order of `language` and `term_language` keys.

Reply:
> ⚠we can't put `ASC` or `DESC` at the end of the orderby clause we modify because `get_terms()` add it with `ASC` value by default.

This doesn't work?
```php
return sprintf( 'tt.taxonomy = "language" DESC, %1$s.term_group, %1$s.term_id', $matches['alias'] );
```

Reply:
> > ⚠we can't put `ASC` or `DESC` at the end of the orderby clause we modify because `get_terms()` add it with `ASC` value by default.
> 
> This doesn't work?
> 
> ```
> return sprintf( 'tt.taxonomy = "language" DESC, %1$s.term_group, %1$s.term_id', $matches['alias'] );
> ```

No! It breaks the languages list order (frontend) and breaks the test I quoted.